app/markets/ActivityChart.tsx

```diff
@@ -0,0 +1,192 @@
+  });
+  const market = marketData?.fixedProductMarketMaker;
+
+  const { data: trades } = useQuery({
+    queryKey: ['getMarketTrades', id],
+    queryFn: async () =>
+      getMarketTrades({
+        fpmm: id,
+        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
+      }),
+  });
+
+  const { data } = useQuery({
+    queryKey: ['getMarketOutcomesHistory', id],
+    queryFn: async (): Promise<DataType[] | undefined> => {
+      if (!trades) return;
+
+      const tradesTimestamp = trades.fpmmTrades.map(trade => trade.creationTimestamp);
+
+      const queryBlocks = `query blockNumberByTimestamp {
+          ${tradesTimestamp.map(
+            timestamp => `_${timestamp}: blocks(where: {timestamp: ${timestamp}}, first: 1) {
+              number
+            }`
+          )}
+        }`;
+
+      const blockNumbers = await request<BlockDataType>(
+        XDAI_BLOCKS_SUBGRAPH_URL,
+        queryBlocks
+      );
+
+      let tradesBlockNumbers = Object.values(blockNumbers).map(value => value[0].number);
+      tradesBlockNumbers = Array.from(new Set(tradesBlockNumbers));
+
+      if (!tradesBlockNumbers) return;
+
+      const queryOutcomeTokenAmounts = `query outcomeTokenAmountsByBlockNumber($id: ID!) {
+          ${tradesBlockNumbers.map(
+            timestamp => `_${timestamp}: fixedProductMarketMaker(id: $id, block: { number: ${timestamp} }) {
+              outcomeTokenAmounts
+            }`
+          )}
+        }`;
+
+      const outcomeTokenAmounts = await request<OutcomesDataType>(
+        OMEN_SUBGRAPH_URL,
+        queryOutcomeTokenAmounts,
+        {
+          id,
+        }
+      );
+
+      return Object.values(outcomeTokenAmounts).map((value, index) => {
+        const outcomes = Object.assign({ 0: '', 1: '' }, value.outcomeTokenAmounts);
+        return { time: tradesTimestamp[index], ...outcomes };
+      });
+    },
+    enabled: !!trades,
+  });
+
+  if (!data || !market)
+    return <div className="h-full w-full animate-pulse rounded-8 bg-outline-low-em" />;
+
+  const TooltipContent = (chartData: DefaultTooltipContentProps<number, string>) => {
+    const { payload } = chartData;
+
+    if (!payload || !payload[0] || !payload[1]) return;
+
+    const total = payload.reduce(
+      (result, entry) => (entry.value ? result + +entry.value : 0),
+      0
+    );
+    const tradeTime = format(fromUnixTime(+payload[0].payload.time), 'HH:mm dd/MM/yy');
+
+    return (
+      <div className="rounded-12 bg-surface-surface-0 p-3 shadow-4">
+        <p>Bets: {data.length}</p>
+        <ul>
+          {payload
+            .sort((a, b) => {
+              if (!a.dataKey || !b.dataKey) return 0;
+              return +a.dataKey - +b.dataKey;
+            })
+            .map((entry, index) => (
+              <li
+                key={index}
+                className={cx({
+                  'text-text-danger-em': entry.dataKey === '0',
+                  'text-text-success-em': entry.dataKey === '1',
```

Comment:
what means '0' and '1'? That's a great opportunity to use a constant to describe this value.

An example
```suggestion
                  'text-text-danger-em': entry.dataKey ===  OUTCOME_NO,
                  'text-text-success-em': entry.dataKey === OUTCOME_NO,
```

Reply:
Change to OUTCOME_0 and OUTCOME_1, there are markets with different outcome names